LIPs/LIP-livepeer_treasury.md

```diff
@@ -0,0 +1,108 @@
+
+Two characteristics which were motivators for this proposal were:
+
+1. Strive for similarities with Livepeer's existing protocol governance. By sticking with these already ratified conventions initially, the community can use said governance to evolve over time towards any updated mechanics.
+2. Bias towards simplicity and the ability to take advantage of existing popular governance tools and frameworks, rather than spend significant time developing, testing, auditing, and iterating on our own bespoke tools.
+
+While important, building governance and treasury management tools are not core activities towards building the world's open video infrastructure, and as such, this proposal is motivated by efficiently taking advantage of existing tools, over theoretically perfect or overoptimized governance mechanisms. 
+
+## Specification
+
+Because this spec covers updates across many areas of the Livepeer protocol, it first attempt to explain a plain English writeup of the intended behaviors encoded into the treasury and associated code updates to capture voting state. But it will be accompanied by two additional assets:
+
+1) A rigorous technical specification of the updates. This will allow an auditor to confirm that the implementation matches the specificly intended behavior.
+2) A reference implementation.
+
+Where the English description contained in this proposal leaves ambiguity, seems at odds with, or omits certain detail, the rigorous technical spec should always take priority.
+
+### Summary
+
+#### Creation of the treasury - The Governor framework
+
+A Livepeer Treasury will be created using the popular [Governor Framework]() as used within many decentralized projects including Compound and Uniswap, and supported by tools such as Tally. In particular, we'll be deploying an instance of a Governor based off of the popular [OpenZeppelin]() implementation. However in order to comply with Livepeer's existing delegated stake weighted governance, our implementation will differ from OpenZeppelin's in that we won’t use the default `GovernorVotes` (”voting power”) and `GovernorCountingSimple` (”tallying”) extensions, but rather implement our custom logic for both.
+
+There are three main parameters to creating a governor instance:
+
+* **Voting Delay** - we propose initializing with a 1 round voting delay, from when proposals are made, until they can be voted on.
+* **Voting Period** - we propose initializing with a 10 round voting period.
+* **Proposal Threshold** - we propose initializing with a 100 LPT minimum staked balance to make a proposal, which matches Livepeer's existing governance.
+
+The Livepeer `LivepeerGovernor` will be the instance of the Governor/Treasury and will be registered with the `Controller` as are all the other smart contracts in the protocol.
```

Comment:
The reason we avoided the `TreasuryGovernor` name is that we already have some expectations in the medium term for this new governor to become the controller owner itself, meaning it would also have control over the protocol and not only the treasury. So avoiding the term Treasury on the high level would be more future-proof.

Another unrelated problem that I hit a couple times in the code is that we now have conflicting declarations with @openzeppelin. After upgrading the library, there are definitions of `IGovernor` and `Governor` that get build errors if they're ever (directly or indirectly) imported in the same file as OZ.

So I'm thinking we could consider 2 directions here:
- Rename the existing governance contracts to something else, maybe `(I)ProtocolGovernor`. This fixes the conflict with OZ and reduces confusion with the new contracts, but might be confusing in terms of how these contracts will stay registered in the controller with a different name (plus communicating the rename in general)
- Pick another name for the new governor contracts that doesn't involve "treasury" but also isn't so generic that it could be confused with the existing protocol governor. Some alternatives: `LivepeerOrg` (my favorite, as in "our DAO"), `LivepeerOrganization`, `LivepeerCommunity`, `CommunityGovernor`, `OrgGovernor`...

Also relevant to note that @dob had mentioned being in conversations with lawyers to know if there are any names we need to avoid for legal liability, so we might need to wait on that to be concluded as well.

WDYT?

Reply:
I suggest that we just pick a name that identifies this as its own thing, doesn't associate it with regulatory risk areas, and doesn't create conflicts with existing OZ or Livepeer contracts. For example: SustainabilityPool. Made that up quickly, but open to any ideas along those lines.

Reply:
In that case, given that context, I think LivepeerGovernor is fine as a name.

As for any code level naming conflicts between the existing LIP-25 Governor and the OZ Governor imports, let's take those discussions into code level PRs.

Reply:
@dob do you think that `LivepeerGovernor` can be associated with any regulatory risk areas? If not, then I think we can commit to that name for now, as it doesn't conflict with any existing contracts.

One might be confused by the difference between `Governor` and `LivepeerGovernor` for a while, but that can be clarified in documentation pieces + in the medium term the existing `Governor` might go away and the confusion be gone

Reply:
I think it's ok for us to name our smart contract LivepeerGovernor without significant risk. So let's go with that.